en/middleware/micrortps.md

```diff
@@ -0,0 +1,271 @@
+
+
+## Architectural overview
+
+![basic example flow](../../assets/middleware/micrortps/basic_example_flow.png)
+
+The main elements of the architecture are the client and agent processes shown in the diagram above.
+
+- The *Client* is PX4 middleware daemon process that runs on the flight controller. It subscribes to uORB topics published by other PX4 components and sends any updates to the *Agent* (via a UART or UDP port). It also receives messages from the *Agent* and publishes them as uORB message on PX4.
+- The *Agent* runs as a daemon process on an offboard computer. It watches for uORB update messages from the *Client* and (re)publishes them over RTPS. It also subscribes to "uORB" RTPS messages from other RTPS applications and forwards them to the *Client*.
+- The *Agent* and *Client* are connected via a serial link (UART) or UDP network. The uORB information is [CDR serialized](https://en.wikipedia.org/wiki/Common_Data_Representation) for sending (*CDR serialization* provides a common format for exchanging serial data between different platforms).
+- The *Agent* and any *FastRTPS* applications are connected via UDP, and may be on the same or another device. In a typical configuration they will both be on the same system (e.g. a development computer, Linux companion computer or compute board), connected to the *Client* over a Wifi link or via USB.
+
+
+## Code generation
+
+The code needed for the *Client*, *Agent*, *CDR serialization/deserialization* of uORB messages, and the definition of the associated RTPS messages is automatically generated when the PX4 Firmware is compiled (and can also be [generated manually](micrortps_manual_code_generation.md)). 
+
+> **Note** [Fast RTPS must be installed](../setup/fast-rtps-installation.md) in order to generate the required code! 
+
+The source code for the *Client* and *Agent* will be generated in **build\__BUILDPLATFORM_/src/modules/micrortps_bridge**, under the **/micrortps_client** and **/micrortps_agent** folders (respectively).
+
+
+### Supported uORB/RTSP messages
+
+The set of uORB topics that can be sent and received by the bridge are listed in the **.cmake** file (**cmake/configs**) for each target platform:
+
+```cmake
+set(config_rtps_send_topics
+  sensor_combined
+   # Add new topic...
+   )
+
+set(config_rtps_receive_topics
+   vehicle_command
+   # Add new topic...
+   )
+```
+
+A uORB serialization function will be generated for each topic in both lists. An IDL file will also be created for each topic from the associated uORB **.msg** files 
+
+> **Note** FastRTSP uses IDL files to define the structure of RTPS messages (in this case, RTPS messages for sending uORB topics). The IDL files are compiled to C++ by the *fastrtpsgen* tool, which is then used by the Agent and other FastRTPS applications.
+
+
+#### uORB serialization code
```

Comment:
@vilhjalmur89 Note that really the serialisation code isn't of that much interest to people wanting to understand the bridge. I've therefore moved it under the generation here as a "for information" rather than part of the detail of the architecture.